syntax/check.rs

```diff
@@ -161,6 +162,24 @@ fn check_type_unique_ptr(cx: &mut Check, ptr: &Ty1) {
     cx.error(ptr, "unsupported unique_ptr target type");
 }
 
+fn check_type_kj_own(cx: &mut Check, ptr: &Ty1) {
+    if let Type::Ident(ident) = &ptr.inner {
+        if cx.types.rust.contains(&ident.rust) {
+            cx.error(ptr, "kj::Own of a Rust type is not supported yet");
+            return;
+        }
+
+        match Atom::from(&ident.rust) {
```

Comment:
what's the point here? to fall through to error? I'd rather we had explicit error flow.